### PR TITLE
Home 5

### PR DESCRIPTION
**Pull request by IgorPrihodko**

### CODE REVIEW

File: src/main/java/controller/product/EditProductServlet.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/editProduct")
+public class EditProductServlet extends HttpServlet {
+
+    private static final ProductService productService = ProductServiceFactory.getInstance();
+    Product product;
```
Comment (Sofasmile) [inline src/main/java/controller/product/EditProductServlet.java]: try doesn't use this variable

File: src/main/java/controller/product/DeleteProductServlet.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        req.setAttribute("id", product.getId());
+        req.setAttribute("title", product.getTitle());
+        req.setAttribute("description", product.getDescription());
+        req.setAttribute("price", product.getPrice());
+        req.getRequestDispatcher("deleteProduct.jsp").forward(req, resp);
+        resp.sendRedirect("/deleteProduct");
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
```
Comment (Sofasmile) [inline src/main/java/controller/product/DeleteProductServlet.java:35]: do we need this method here?
maybe better use only get method?

File: src/main/java/controller/user/DeleteUserServlet.java [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+        User user = userService.getById(id);
+        req.setAttribute("id", user.getId());
+        req.setAttribute("email", user.getEmail());
+        req.setAttribute("password", user.getPassword());
+        req.getRequestDispatcher("deleteUser.jsp").forward(req, resp);
+        resp.sendRedirect("/deleteUser");
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
```
Comment (Sofasmile) [inline src/main/java/controller/user/DeleteUserServlet.java:34]: the same

File: src/main/java/controller/user/EditUserServlet.java [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@WebServlet(value = "/editUser")
+public class EditUserServlet extends HttpServlet {
+
+    private static final UserService userService = UserServiceFactory.getInstance();
+    User user;
```
Comment (Sofasmile) [inline src/main/java/controller/user/EditUserServlet.java]: -)

File: src/main/java/dao/product/ProductDao.java [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+import model.Product;
+
+import java.util.List;
+
+public interface ProductDao {
+
+    Long createID();
+    void addProduct(Product product);
+    void removeProduct(Long id);
+    Product getById (Long id);
```
Comment (Sofasmile) [inline src/main/java/dao/product/ProductDao.java]: try to return optional everywhere

File: src/main/java/dao/product/ProductDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+    @Override
+    public void addProduct(Product product) {
+        product.setId(createID());
+        PRODUCTS.add(product);
+    }
+
+    @Override
+    public void removeProduct(Long id) {
+        for (Product product : PRODUCTS) {
```
Comment (Sofasmile) [inline src/main/java/dao/product/ProductDaoImpl.java]: try to use stream api in

File: src/main/java/service/user/UserServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+import dao.user.UserDao;
+import factory.UserDaoFactory;
+import model.User;
+
+import java.util.List;
+
+public class UserServiceImpl implements UserService {
+
+    private  static final UserDao userDao = UserDaoFactory.getInstance();
```
Comment (Sofasmile) [inline src/main/java/service/user/UserServiceImpl.java]:
```suggestion
    private static final UserDao userDao = UserDaoFactory.getInstance();
```

File: src/main/java/dao/user/UserDaoImpl.java [hunk 8]
```diff
@@ -0,0 +1,57 @@
+package dao.user;
+
+import model.User;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class UserDaoImpl implements UserDao {
+
+    private static final List<User> USERS = new ArrayList<>();
```
Comment (Sofasmile) [inline src/main/java/dao/user/UserDaoImpl.java]: add logger everywhere

File: src/main/java/controller/product/DeleteProductServlet.java (hunk 2 above)
Comment (IgorPrihodko) [reply src/main/java/controller/product/DeleteProductServlet.java:35]: is it ok to use doGet for posting changes to DB?

File: src/main/java/controller/product/EditProductServlet.java (hunk 1 above)
Comment (IgorPrihodko) [reply src/main/java/controller/product/EditProductServlet.java]: fixed

File: src/main/java/controller/user/DeleteUserServlet.java (hunk 3 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/DeleteUserServlet.java:34]: asked before

File: src/main/java/controller/user/EditUserServlet.java (hunk 4 above)
Comment (IgorPrihodko) [reply src/main/java/controller/user/EditUserServlet.java]: fixed

File: src/main/java/dao/product/ProductDao.java (hunk 5 above)
Comment (IgorPrihodko) [reply src/main/java/dao/product/ProductDao.java]: fixed

File: src/main/java/dao/product/ProductDaoImpl.java (hunk 6 above)
Comment (IgorPrihodko) [reply src/main/java/dao/product/ProductDaoImpl.java]: fixed

File: src/main/java/dao/user/UserDaoImpl.java (hunk 8 above)
Comment (IgorPrihodko) [reply src/main/java/dao/user/UserDaoImpl.java]: done

File: src/main/java/service/user/UserServiceImpl.java (hunk 7 above)
Comment (IgorPrihodko) [reply src/main/java/service/user/UserServiceImpl.java]: fixed

**Pull request merged**